bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v17/PsbtRpcTest.scala

```diff
@@ -0,0 +1,223 @@
+package org.bitcoins.rpc.v17
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.testkit.TestKit
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.currency.Bitcoins
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.ScriptSignature
+import org.bitcoins.core.protocol.transaction.{
+  TransactionConstants,
+  TransactionInput,
+  TransactionOutPoint
+}
+import org.bitcoins.rpc.BitcoindRpcTestUtil
+import org.bitcoins.rpc.client.common.RpcOpts.AddNodeArgument
+import org.bitcoins.rpc.client.v17.BitcoindV17RpcClient
+import org.bitcoins.rpc.jsonmodels.{FinalizedPsbt, NonFinalizedPsbt}
+import org.scalatest.{AsyncFlatSpec, BeforeAndAfterAll}
+
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration.DurationInt
+
+class PsbtRpcTest extends AsyncFlatSpec with BeforeAndAfterAll {
+  implicit val system: ActorSystem = ActorSystem(
+    "BitcoindV17RpcClientTest_ActorSystem")
+  implicit val m: ActorMaterializer = ActorMaterializer()
+  implicit val ec: ExecutionContext = m.executionContext
+  implicit val networkParam: NetworkParameters = BitcoindRpcTestUtil.network
+
+  val client = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+  val otherClient = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+  val thirdClient = new BitcoindV17RpcClient(BitcoindRpcTestUtil.v17Instance())
+
+  override protected def beforeAll(): Unit = {
+    import org.bitcoins.rpc.BitcoindRpcTestConfig.DEFAULT_TIMEOUT
+
+    val startF =
+      BitcoindRpcTestUtil.startServers(Vector(client, otherClient, thirdClient))
+    Await.result(startF, DEFAULT_TIMEOUT)
+
+    val pairs = List((client, otherClient), (client, thirdClient))
+    val addNodesF = Future.sequence(pairs.map {
+      case (first, second) =>
+        first.addNode(second.getDaemon.uri, AddNodeArgument.Add)
+    })
+
+    Await.result(addNodesF, DEFAULT_TIMEOUT)
+
+    pairs.foreach {
+      case (first, second) =>
+        BitcoindRpcTestUtil.awaitConnection(first, second, duration = 1.second)
+    }
+
+    Await.result(client.generate(200), DEFAULT_TIMEOUT)
```

Comment:
In general it's a bad idea to do `Await.result` like this, we should aggregate the futures and then Await if we must. We probably don't need to await here though, we should just have 

```scala
val clientF = client.generate(200).map(_ => client)
```